example/rails_app/lib/event_handler.rb

```diff
@@ -2,6 +2,6 @@ module EventHandler
   extend self
 
   def handle_event event
-    puts "RailsApp::EventHandler - Consumed event: #{event.name} with payload: #{event.payload}"
+    puts "RailsApp::EventHandler - Consumed event: #{event.metadata[:name]} with payload: #{event.payload}"
```

Comment:
I think a helper method for the name, that delegates to the metadata, would be nice yes.